### PR TITLE
Fix more DOMRect constructor tests and add new tests

### PR DESCRIPTION
**Pull request by tschneidereit**

### CODE REVIEW

**Comment by zcorpan** [comment]
Can you make it DRY?

File: geometry-1/DOMRect-001.html [hunk 1]
```diff
@@ -60,7 +60,7 @@
         },'testConstructorNegativeWidthHeight');
         test(function() {
             checkDOMRect(new DOMRect(0, 0, undefined, 4),
-                { x: 0, y: 0, width: NaN, height: 4, top: 0, right: 0, bottom: 4, left: 0 });
+                { x: 0, y: 0, width: 0, height: 4, top: 0, right: 0, bottom: 4, left: 0 });
```
Comment (zcorpan) [inline geometry-1/DOMRect-001.html]: I think NaN was right.
http://heycam.github.io/webidl/#es-unrestricted-double
http://people.mozilla.org/~jorendorff/es6-draft.html#sec-tonumber

Comment (tschneidereit) [inline geometry-1/DOMRect-001.html]: I think the default value should be used here, see [4.5.1.1., step 10.4.](http://heycam.github.io/webidl/#es-interface-call).

**Comment by tschneidereit** [comment]
> Can you make it DRY?

You're right, of course. Will do.

**Comment by tschneidereit** [comment]
@zcorpan, made it more DRY. See my previous comment regarding the `NaN` thing.

File: geometry-1/DOMRect-001.html (hunk 1 above)
Comment (zcorpan) [inline geometry-1/DOMRect-001.html]: Yeah, @heycam confirmed your reading is right.

**Comment by zcorpan** [comment]
LGTM

**Pull request merged**